src/migrate/index.js

```diff
@@ -121,22 +131,84 @@ export default class Migrator {
     });
   }
 
+  _createMigrationLockTable(tableName) {
+    return this.knex.schema.createTable(tableName, function(t) {
+      t.boolean('is_locked');
+    });
+  }
+
+  _getLockTableName(tableName) {
+    return tableName + '_lock';
+  }
+
+  _isLocked(trx) {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName)
+      .transacting(trx)
+      .forUpdate()
+      .select('*')
+      .then(data => {
+        // This is to handle the different databases
+        // data[0].is_locked == false will also work but
+        // JSHint requires === which won't do triple equals
+        if (data[0].is_locked !== false &&
```

Comment:
This looks really error prone... would this be easier if `is_locked` column type would be e.g. integer or even string?